File: CS50x_computer_science/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # ASK USER FOR IMPUT
    text = get_string("Text: ")

    # EXECUTE FUNCTIONS
    textLength = len(text)
    textWords = count_words(text, textLength)
    textSentences = count_senteces(text, textLength)
    textLetters = count_letters(text, textLength)

    # DEFINE L AND S
    L = (textLetters / textWords)*100
    S = (textSentences / textWords)*100

    # PRINT GRADE
    print_grade(L, S)


# FUNCTION TO COUNT LETTERS
def count_letters(text, textLength):
    i = 0
    textLetters = 0
    while i < textLength:
        if text[i].isalpha() == True:
            textLetters += 1
            i += 1
        else:
            i += 1
    return textLetters


# FUNCTION TO COUNT WORDS
def count_words(text, textLength):
    i = 0
    textWords = 1
    while i < textLength:
        if text[i].isspace() == True:
            textWords += 1
            i += 1
        else:
            i += 1
    return textWords


# FUNCTION TO COUNT SENTENCES
def count_senteces(text, textLength):
    i = 0
    textSentences = 0
    while i < textLength:
        if text[i] == "." or text[i] == "?" or text[i] == "!":
            textSentences += 1
            i += 1
        else:
            i += 1
    return textSentences


# FUNCTION TO PRINT GRADE
def print_grade(L, S):
    grade = round(0.0588 * L - 0.29 * S - 15.8)

    if grade < 1:
        print("Before Grade 1")
    if grade > 1 and grade < 16:
        print(f"Grade {grade}")
    if grade > 16:
        print("Grade 16+")


main()

File: CS50x_conputer_science/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text, int tl);
int count_words(string text, int tl);
int count_sentences(string text, int tl);
int main(void)
{
    //GET TEXT
    string text = get_string("TEXT: ");

    //GET TEXT LENGHT
    float tl = strlen(text);

    //GET NUMBEROF LETTERS
    float nl = count_letters(text, tl);

    //GET NUMBER OF WORDS
    float nw = count_words(text, tl);

    //GET NUMBER OF SENTENCES
    float ns = count_sentences(text, tl);

    //CALCULATE L
    float l = (nl * 100) / nw;

    //CALCULATE S
    float s = (ns * 100) / nw;

    //CALCULATE GRADE
    float index = (0.0588 * l - 0.296 * s - 15.8);

    int grade = round(index);

    //PRINT GRADE
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

//FUNCIONS


//GET NUMBEROF LETTERS
int count_letters(string text, int tl)
{
    int i = 0, count = 0;
    do
    {
        if (isalpha(text[i]))
        {
            count++;
            i++;
        }
        else
        {
            i++;
        }
    }
    while (i < tl);
    return count;
}


//GET NUMBER OF WORDS
int count_words(string text, int tl)
{
    int i = 0, count = 0;
    do
    {
        if (isspace(text[i]))
        {
            count++;
            i++;
        }
        else
        {
            i++;
        }
    }
    while (i < tl);
    return count + 1;
}

//GET NUMBER OF SENTENCES
int count_sentences(string text, int tl)
{
    int i = 0, count = 0;
    char s1 = '.', s2 = '!', s3 = '?';
    do
    {
        if (text[i] == s1 || text[i] == s2 || text[i] == s3)
        {
            count++;
            i++;
        }
        else
        {
            i++;
        }
    }
    while (i < tl);
    return count;
}